### PR TITLE
(feat): add --console-url testing flag for OAuth against alternative consoles

### PR DESCRIPTION
**Pull request by ChiragAgg5k**
## What

Adds a centralized testing-flag registry and its first flag, `--console-url` / `MCP_CONSOLE_URL`, so testers can run the hosted OAuth flow through a pre-release console (e.g. the new console at `new.appwrite.io`) instead of the production console.

## Why

The Appwrite authorization server hard-codes its login/consent redirect to the production console (`_APP_CONSOLE_OAUTH2_AUTHORIZATION_URL` server-side), so testers had no way to exercise MCP OAuth against the new console.

## How

- **`flags.py` (new)** — central registry: each flag declares CLI name, env var, and help once, and automatically gets the `--<name>` argument and `$<ENV>` variable. The environment is the runtime source of truth; values are read at request time so tests toggle them via `os.environ`.
- **`auth.py`** — with the flag set, protected-resource metadata names the MCP server itself as the authorization server, and the mirrored discovery document rewrites only `authorization_endpoint` to a local proxy. Token/registration/JWKS endpoints stay upstream, so token validation is unchanged.
- **`http_app.py`** — new `/oauth2/authorize` proxy (404 when the flag is off): forwards the request to the real authorize endpoint (keeping upstream validation), rewrites the `…/console/oauth2/consent` redirect to `<console-url>/oauth2/consent`, and passes error redirects / non-redirect responses through verbatim. Also adds a `/.well-known/openid-configuration` alias for clients that discover via OIDC.
- **`docs/flags.md` (new)** — how to add a flag, enable it (CLI or env), and verify it (curl checks per hop + full `claude mcp add` flow). Linked from AGENTS.md.

## Verification

Verified end to end against `new.appwrite.io`: discovery → local authorize proxy → new-console sign-in + consent → authorization code → PKCE token exchange (with RFC 8707 `resource`) → authenticated MCP `initialize`, `tools/list`, and `tools/call` against the flag-enabled server.

| New console sign-in | Consent | 
| --- | --- |
| reached via the local authorize proxy | approved, code delivered to local callback |

> **Found during testing (separate issue, new console repo):** the new console's consent page drops the RFC 8707 `resource` query parameter when replaying the authorize call, so grants are created without resource indicators and spec-compliant MCP clients fail token exchange with `invalid_target`. Reproduced with this flag on vs. off; the full chain works when `resource` is preserved.

Unit tests: 173 pass; ruff/black/pyright clean.

### CODE REVIEW

File: src/mcp_server_appwrite/http_app.py [hunk 1]
```diff
@@ -324,11 +328,56 @@ async def authorization_server_metadata_endpoint(request: Request) -> JSONRespon
...
+        upstream = f"{upstream}?{request.url.query}"
+
+    async with httpx.AsyncClient(timeout=10.0, follow_redirects=False) as client:
+        resp = await client.get(upstream)
+
+    location = resp.headers.get("location")
+    if resp.status_code in _REDIRECT_STATUSES and location:
+        parts = urlsplit(location)
+        if parts.path.rstrip("/").endswith("/oauth2/consent"):
+            location = f"{console_url()}/oauth2/consent?{parts.query}"
```
[the remark this answers is not in the document]
Comment (ChiragAgg5k) [reply src/mcp_server_appwrite/http_app.py]: Fixed in ce975e8 — query-less consent redirects no longer get a bare `?`, CLI flags default to None so `--console-url ""` clears an env-set flag, and both edge cases (plus the flag-unset 404) are covered by tests.

File: src/mcp_server_appwrite/flags.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            help=f"Testing flag: {flag.help} (default ${flag.env}).",
+        )
+
+
+def apply_cli_args(args: argparse.Namespace) -> None:
+    """Write parsed CLI flag values back to their environment variables."""
+    for flag in FLAGS:
+        raw = getattr(args, flag.name.replace("-", "_"), "")
+        if raw:
+            os.environ[flag.env] = raw
```
[the remark this answers is not in the document]
Comment (ChiragAgg5k) [reply src/mcp_server_appwrite/flags.py:76]: Fixed in ce975e8 — query-less consent redirects no longer get a bare `?`, CLI flags default to None so `--console-url ""` clears an env-set flag, and both edge cases (plus the flag-unset 404) are covered by tests.

File: tests/unit/test_http_app.py [hunk 3]
```diff
@@ -313,6 +315,106 @@ def test_app_routes_include_legacy_discovery_paths(self):
...
+        factory, _client = self._fake_httpx_client(upstream)
+        with mock.patch("mcp_server_appwrite.http_app.httpx.AsyncClient", factory):
+            response = asyncio.run(
+                oauth_authorize_proxy_endpoint(self._request("client_id=abc"))
+            )
+        self.assertEqual(response.status_code, 303)
+        self.assertEqual(
+            response.headers["location"],
+            "http://localhost:33418/callback?error=invalid_scope",
+        )
```
[the remark this answers is not in the document]
Comment (ChiragAgg5k) [reply tests/unit/test_http_app.py:429]: Fixed in ce975e8 — query-less consent redirects no longer get a bare `?`, CLI flags default to None so `--console-url ""` clears an env-set flag, and both edge cases (plus the flag-unset 404) are covered by tests.
